### ISSUE TITLE
DHCP client fails to renew IP address on WAN interface link flapping

### ISSUE THREAD

**Issue by shteryana** [opnsense/core#2372]
DHCP client fails to renew IP address on WAN interface link flapping with Opnsense 18.1.6

The Opnsense system is configured to obtain WAN IP address via DHCP. On link down & link up event on the WAN interface we are observing the following system log and the system remains without IP address on the WAN interface - 
```
Apr 17 23:26:28     opnsense: /usr/local/etc/rc.linkup: The command '/sbin/route add -'inet' default 'xx.xx.xx.xx'' returned exit code '1', the output was 'route: writing to routing socket: Network is unreachable add net default: gateway xx.xx.xx.xx fib 0: Network is unreachable'
Apr 17 23:26:28     opnsense: /usr/local/etc/rc.linkup: ROUTING: setting IPv4 default route to xx.xx.xx.xx
Apr 17 23:26:28     opnsense: /usr/local/etc/rc.linkup: ROUTING: no IPv6 default gateway set, trying wan on 'igb0' ()
Apr 17 23:26:28     opnsense: /usr/local/etc/rc.linkup: ROUTING: creating /tmp/igb0_defaultgw
Apr 17 23:26:28     opnsense: /usr/local/etc/rc.linkup: ROUTING: removing /tmp/igb0_defaultgwv6
Apr 17 23:26:28     opnsense: /usr/local/etc/rc.linkup: ROUTING: removing /tmp/igb0_defaultgw
Apr 17 23:26:28     opnsense: /usr/local/etc/rc.linkup: ROUTING: entering configure using 'wan'
Apr 17 23:26:28     opnsense: /usr/local/etc/rc.linkup: The command '/sbin/dhclient -c /var/etc/dhclient_wan.conf igb0 > /tmp/igb0_output 2> /tmp/igb0_error_output' returned exit code '1', the output was ''
Apr 17 23:26:27     opnsense: /usr/local/etc/rc.linkup: HOTPLUG: Configuring interface wan
Apr 17 23:26:27     opnsense: /usr/local/etc/rc.linkup: DEVD Ethernet attached event for wan
Apr 17 23:26:27     opnsense: /usr/local/etc/rc.linkup: Clearing states to old gateway xx.xx.xx.xx.
```

We have been running the following patch for a few days and it seems to resolve the issue for us on link flapping events
```
--- /usr/local/etc/inc/interfaces.inc.orig	2018-04-18 11:09:38.435640000 +0300
+++ /usr/local/etc/inc/interfaces.inc	2018-04-18 11:16:55.937207000 +0300
@@ -3499,7 +3499,7 @@
     kill_dhclient_process($wanif);
 
     /* fire up dhclient */
-    mwexec("/sbin/dhclient -c /var/etc/dhclient_{$interface}.conf {$wanif} > /tmp/{$wanif}_output 2> /tmp/{$wanif}_error_output");
+    mwexec("/sbin/dhclient -c /var/etc/dhclient_{$interface}.conf {$wanif} > /tmp/{$wanif}_output");
 }
 
 function DHCP_Config_File_Advanced($interface, $wancfg, $wanif)
```

**Comment by fichtner** [comment]
Weird, but I will take a look. Maybe we can get rid of the output redirect completely.

**Comment by fichtner** [comment]
200f92b75 should do the trick, the output of mwexec is always folded (stderr to stdout and used in the error message when the return value is non-zero)

**Comment by fichtner** [comment]
Shipped in 18.1.7.

**Comment by fichtner** [comment]
There a new patch in 65186994744 but it doesn't change much again. During interface down/up it seems that dhclient is stuck and cannot be killed, which is really unfortunate because they do not implement SIGHUP so we are stuck with a blocking process that won't do anything anymore.
